paddle/framework/framework.proto

```diff
@@ -143,3 +143,13 @@ message BlockDesc {
 // https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md
 // for more details.
 message ProgramDesc { repeated BlockDesc blocks = 1; }
+
+message OpPlacement {
+  optional string name = 1;
+  optional string device = 2;
```

Comment:
Why not put `device` field in `ProgramDesc` directly?

Reply:
Maybe we also need to allow users the specify the device information by two approaches:
- device ID such as `CPU:0/GPU:0`.
- The maximal device count such as `CPU:{5}`.

Reply:
The `ProgramDesc` is used to specify the information from the user, since currently we don't have API to do that, we probably should not put that information into `ProgramDesc`.

In the future when we have that API we can add it to `ProgramDesc`.

Reply:
It's fine to add it to `ProgramDesc` since we are not using this field for now, so then we don't need further changes to the protobuf files.

Reply:
@typhoonzero I think `ProgramDesc` and `ExecutionPlan` are used for different purposes, `ProgramDesc` is the output from Python, specifying what the user need. `ExecutionPlan` is the input and output for IR optimizers, and input for executor. So they better be two separate entities.

Since there are two entities: `ProgramDesc` and `ExecutionPlan`, since the device placement is about optimization, and not about what the user specified, it probably should be in `ExecutionPlan` but not `ProgramDesc`.

In the future when we want enable the user to configure which device an OP runs, we can put the field indicating device in `ProgramDesc`.